src/__smoke-test__/smoke-test.spec.ts

```diff
@@ -0,0 +1,446 @@
+import { test, expect } from '@playwright/test';
+import {
+    HTTP_EVENT_TYPE,
+    JS_ERROR_EVENT_TYPE,
+    LCP_EVENT_TYPE,
+    FID_EVENT_TYPE,
+    PERFORMANCE_NAVIGATION_EVENT_TYPE,
+    PERFORMANCE_RESOURCE_EVENT_TYPE,
+    PAGE_VIEW_EVENT_TYPE,
+    SESSION_START_EVENT_TYPE
+} from '../plugins/utils/constant';
+
+// Secrete constant values retrieved from GitHub
+const endpoint = '${process.env.ENDPOINT}';
+const monitorId = '${process.env.MONITOR_ID}';
+const url = '${process.env.SMOKE_URL}';
+
+function parseBatches(requestData) {
+    const output = [];
+    for (let i = 0; i < requestData.length; i++) {
+        const parsed = JSON.parse(requestData[i]).RumEvents;
+        for (let j = 0; j < parsed.length; j++) {
+            output.push(parsed[j]);
+        }
+    }
+    return output;
+}
+
+function addEventsToList(parsedPayload, eventList) {
+    for (let i = 0; i < parsedPayload.length; i++) {
+        eventList.push(parsedPayload[i].id);
+    }
+}
+
+function getEventsByType(requestBody, eventType) {
+    return requestBody.filter((e) => e.type === eventType);
+}
+
+function filterSuccessfulResponse(response, requestBody) {
+    // filter out irrelevant, unsuccessful network request/response
+    const request = response.request();
+    if (
+        request &&
+        response.status() === 200 &&
+        request.method() === 'POST' &&
+        response.url() === endpoint + monitorId + '/' &&
+        request.postData()
+    ) {
+        requestBody.push(request.postData());
+    }
```

Comment:
suggestion: Array functions like `filter` and `map` may make this a bit cleaner.

Reply:
Good call, will reflect in next revision.

Reply:
Response is an object, so array functions are not applicable. Removed two conditions, `request` and `request.postData()` as a `response` object will always have `request` object and a `request` object will always have `postData()`